### ISSUE TITLE
make sure `-std=c++17` is well supported

### ISSUE THREAD

**Issue by h-2** [seqan/seqan#2201]
- [x] add at least one build to nightlies
  - [x] add at least one build to travis
  - [x] fix issues

**Comment by h-2** [comment]
It already looks like simd_tests are failing with c++17, but maybe that code will be replaced by #2118 anyways?

**Comment by h-2** [comment]
These tests fail locally with `std=c++17`:
```
          3 - test_test_align_simd_sse4_host (Not Run)
         59 - test_test_journaled_string_tree (Not Run)
         61 - test_journaled_string_tree_traverser (Not Run)
         62 - test_journaled_string_tree_find (Not Run)
        260 - test_demo_tutorial_journaled_string_tree_journaled_string_tree_base (Failed)
        261 - test_demo_tutorial_journaled_string_tree_journaled_string_tree_base_assignment1 (Failed)
```

Since this is "your turf", can you have a alook, @rrahn? Thanks!

**Comment by h-2** [comment]
fixed by #2209
